### ISSUE TITLE
`reloadDates[]` breaks scrolling sometimes. 

### ISSUE THREAD

**Issue by ahmed-sal** [patchthecode/JTAppleCalendar#556]
Firstly,

Thanks Jay for this awesome library. I'm starting to become a big user and hopefully will help contribute as I get more involved in my app. 

## The problem 
This is just a heads up for devs: 
In version `7.0.6` if you use `calendarView.reloadDates[<some_dates>]`, it results in breaking selection functionality.

Steps to reproduce:
1. reload some dates using `calendarView.reloadDates[<some_dates>]`
2. scroll to a different month (my case: horizontal scrolling, one month, paging and 6 rows)
3. Select any date on the new month
4. Observe that the calendar automatically scrolls back to the previous month (or whichever month you reloaded with in step 1).
5. Also observe that if you breakpoint `didSelectCell:forDate:`, the cell is `nil`.

## The solution 
The fix for that is to use this function instead
`reloadData(anchorDate:completion:)`. After a long day of searching, this fixed my issue.

**Comment by patchthecode** [comment]
Thanks man. 
Yes. Anchor date prevents the calendar from resetting back to the start.

however, `calendarView.reloadDates[<some_dates>]` should not have behaved this way.
I think I have fixed it on the master branch so that it does not behave like this.
I have updated the changes on master branch and have tracked them here -> https://github.com/patchthecode/JTAppleCalendar/issues/553

Thanks again.
